Quadriga/src/main/java/edu/asu/spring/quadriga/service/network/impl/NetworkTransformationManager.java

```diff
@@ -75,6 +75,71 @@ public ITransformedNetwork getTransformedNetworkOfProject(String projectId)
         if (networkList == null) {
             return null;
         }
+        
+        ITransformedNetwork transformedNetwork = getTransformedNetworkusingNetworkList(networkList);
+        
+        return transformedNetwork;
+    }
+
+    @Override
+    public ITransformedNetwork getSearchTransformedNetwork(String projectId, String conceptId)
+        throws QuadrigaStorageException {
+        // get the transformed network and search for the concept id
+        List<INetwork> networkList = getNetworkList(projectId);
+
+        if (networkList == null) {
+            return null;
+        }
+
+        // get the transformed network of all the networks in a project
+        ITransformedNetwork transformedNetwork = getTransformedNetworkusingNetworkList(networkList);
+        // create finalnetwork using conceptId 
+        ITransformedNetwork finalTransformedNetwork = getfinalTransformedNetwork(transformedNetwork,conceptId);
+
+        return finalTransformedNetwork;
+    }
+
+    private List<INetwork> getNetworkList(String projectId) throws QuadrigaStorageException {
+        List<INetwork> networkList;
+        try {
+            networkList = networkManager.getNetworksInProject(projectId);
+        } catch (QuadrigaStorageException e) {
+            throw new QuadrigaStorageException("Database error while getting networks of a project" +
+                    " with id: " + projectId, e);
+        }
+
+        return networkList;
+    }
+    
+    
+    @Override
+    public ITransformedNetwork getSearchTransformedNetworkMultipleProjects(List<String> projectIds, String conceptId)
+        throws QuadrigaStorageException {
+    	// get the transformed network and search for the concept id
+        List<INetwork> networkList = new ArrayList<>();
+        
+        for(String projectId : projectIds){
+        	List<INetwork> networks = getNetworkList(projectId);
+        	if(networks != null){
+        		networkList.addAll(networks);
+        	}
+        }
+        
+        if (networkList.size()==0) {
```

Comment:
why return null and not an empty list? Would make much more sense. If there are no networks, then the list is empty. Null usually indicates something else.